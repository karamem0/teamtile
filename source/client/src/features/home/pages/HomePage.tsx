//
// Copyright (c) 2021-2024 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

import React from 'react';

import { Event } from '../../../types/Event';

import Presenter from './HomePage.presenter';

function HomePage() {

  const handleLinkClick = React.useCallback((_?: Event, data?: string) => {
    switch (data) {
      case 'GitHub':
        window.open('https://github.com/karamem0/teamtile', '_blank');
        break;
      case 'TermsOfUse':
        window.open('https://github.com/karamem0/teamtile/blob/main/TERMS_OF_USE.md', '_blank');
        break;
      case 'PrivacyPolicy':
        window.open('https://github.com/karamem0/teamtile/blob/main/PRIVACY.md', '_blank');
        break;
      default:
        break;
    }
  }, []);

  return (
    <Presenter onLinkClick={handleLinkClick} />
  );

}

export default HomePage;
