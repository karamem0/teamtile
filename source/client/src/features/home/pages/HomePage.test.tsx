//
// Copyright (c) 2022 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

import React from 'react';

import { render, screen } from '@testing-library/react';

import HomePage from './HomePage.presenter';

test('create shapshot', async () => {
  const params = {};
  render(<HomePage {...params} />);
  expect((await screen.findAllByText(/^.*$/))[0]).toMatchSnapshot();
});
