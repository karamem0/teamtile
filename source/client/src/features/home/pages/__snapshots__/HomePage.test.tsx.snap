// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match the snapshot 1`] = `
<DocumentFragment>
  .emotion-0 {
  min-height: 100vh;
  line-height: 1.25em;
  background-color: #f5f5f5;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-flow: column;
  -webkit-flex-flow: column;
  -ms-flex-flow: column;
  flex-flow: column;
}

.emotion-2 {
  display: grid;
  grid-template-rows: auto;
  grid-template-columns: auto;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: end;
  -ms-flex-pack: end;
  -webkit-justify-content: end;
  justify-content: end;
  width: 100%;
  height: 2rem;
  padding: 0 1rem;
}

.emotion-3 {
  font-size: 1rem;
  line-height: 1rem;
}

.emotion-4 {
  display: grid;
  grid-template-rows: auto;
  grid-template-columns: auto;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  width: 100%;
  padding: 2rem;
  background-color: #4c6978;
}

@media (width >= 960px) {
  .emotion-4 {
    padding: 4rem 2rem;
  }
}

.emotion-5 {
  display: grid;
  grid-template-rows: auto;
  grid-template-columns: auto;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

@media (width >= 960px) {
  .emotion-5 {
    grid-template-rows: auto;
    grid-template-columns: 1fr 1fr;
  }
}

.emotion-6 {
  display: grid;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-7 {
  font-size: 3rem;
  font-weight: 700;
  line-height: calc(3rem * 1.25);
  color: #ffffff;
  text-align: center;
}

.emotion-8 {
  color: #ffffff;
}

.emotion-9 {
  height: auto;
}

.emotion-10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-flow: column;
  -webkit-flex-flow: column;
  -ms-flex-flow: column;
  flex-flow: column;
  grid-gap: 1rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  padding: 2rem 0;
}

.emotion-11 {
  font-size: 2rem;
  font-weight: 700;
  line-height: calc(2rem * 1.25);
  text-align: center;
}

.emotion-12 {
  text-align: center;
}

.emotion-13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-flow: column;
  -webkit-flex-flow: column;
  -ms-flex-flow: column;
  flex-flow: column;
  grid-gap: 0.5rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  list-style: none;
}

.emotion-14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-flow: row;
  -webkit-flex-flow: row;
  -ms-flex-flow: row;
  flex-flow: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  padding: 2rem 0;
}

.emotion-15 {
  padding: 0 0.25rem;
}

<div
    class="fui-FluentProvider fui-FluentProviderr0 ___jdtuxv0_17k0bs4 f19n0e5 fxugw4r f1o700av fk6fouc fkhj508 figsok6 f1i3iumi"
    dir="ltr"
  >
    <div
      class="emotion-0"
    >
      <div
        class="emotion-1"
      >
        <header
          class="emotion-2"
        >
          <a
            class="fui-Button r1alrhcs ___eduuxo0_1tclg4b f1c21dwh f1p3nwhy f11589ue f1q5o8ev f1pdflbu fkfq4zb fjxutwb f1s2uweq fr80ssc f1ukrpxl fecsdlb f139oj5f ft1hn21 fuxngvv fwiml72 f1h0usnq fs4ktlq f16h9ulv fx2bmrt f1fg1p5m f1dfjoow f1j98vj9 f1tme0vf f4xjyn1 f18onu3q f9ddjv3"
            role="button"
            tabindex="0"
          >
            <span
              class="fui-Button__icon rywnvv2 ___963sj20_1qpm2bl f1nizpg2"
            >
              <svg
                class="emotion-3"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M11.9989871,1 C5.92550416,1 1,5.92482888 1,12.0003376 C1,16.8603395 4.15153934,20.9829338 8.52263728,22.4374904 C9.0729918,22.5387827 9.27355045,22.199116 9.27355045,21.9073943 C9.27355045,21.6467356 9.2640965,20.954572 9.25869425,20.0368642 C6.19899322,20.7013414 5.55342398,18.5620492 5.55342398,18.5620492 C5.0530403,17.2911692 4.33183953,16.9528531 4.33183953,16.9528531 C3.33309801,16.2708186 4.40747107,16.2843243 4.40747107,16.2843243 C5.51155652,16.3619816 6.09229872,17.4181221 6.09229872,17.4181221 C7.07348292,19.0988981 8.66714755,18.6133706 9.2938089,18.3317781 C9.39375058,17.6213819 9.67804414,17.1365297 9.99205009,16.86169 C7.54955646,16.5841493 4.98146045,15.6401056 4.98146045,11.4249977 C4.98146045,10.224347 5.41026428,9.24181221 6.11390773,8.47334172 C6.00046042,8.19512569 5.62297799,7.07618404 6.22195279,5.56220265 C6.22195279,5.56220265 7.14506277,5.26642929 9.24653918,6.68992296 C10.12373,6.44547101 11.0650726,6.32392032 12.0003376,6.31919335 C12.9349274,6.32392032 13.8755947,6.44547101 14.7541361,6.68992296 C16.8542619,5.26642929 17.7760214,5.56220265 17.7760214,5.56220265 C18.3763467,7.07618404 17.9988643,8.19512569 17.8860923,8.47334172 C18.5910863,9.24181221 19.0165137,10.224347 19.0165137,11.4249977 C19.0165137,15.6509101 16.444366,16.5807729 13.9944443,16.8529114 C14.3888087,17.192578 14.7406305,17.863808 14.7406305,18.890236 C14.7406305,20.3603241 14.7271248,21.5467939 14.7271248,21.9073943 C14.7271248,22.2018171 14.9256576,22.5441849 15.4834403,22.4368151 C19.8511618,20.9788821 23,16.8589889 23,12.0003376 C23,5.92482888 18.0744958,1 11.9989871,1"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
            GitHub
          </a>
        </header>
        <section
          class="emotion-4"
        >
          <div
            class="emotion-5"
          >
            <div
              class="emotion-6"
            >
              <h1
                class="fui-Text emotion-7 ___c56hd20_1xy0w11 fk6fouc fkhj508 f1i3iumi figsok6 fpgzoln f1w7gpdv f6juhto f1gl81tg f2jf649"
              >
                Teamtile
              </h1>
              <span
                class="fui-Text emotion-8 ___c56hd20_1xy0w11 fk6fouc fkhj508 f1i3iumi figsok6 fpgzoln f1w7gpdv f6juhto f1gl81tg f2jf649"
              >
                Tile your teams for quick access.
              </span>
            </div>
            <img
              alt="Teamtile"
              class="fui-Image emotion-9 ___18apvp7_1wq7hmk fj3muxo f1akhkt f1aperda f1lxtadh f1fabniw f1ewtqcl f14t3ns0 f1kle4es f14xojzb f1l02sjl fly5x3f"
              src="/assets/screenshots/001.png"
            />
          </div>
        </section>
        <section
          class="emotion-10"
        >
          <h2
            class="fui-Text emotion-11 ___c56hd20_1xy0w11 fk6fouc fkhj508 f1i3iumi figsok6 fpgzoln f1w7gpdv f6juhto f1gl81tg f2jf649"
          >
            Features
          </h2>
          <span
            class="fui-Text emotion-12 ___c56hd20_1xy0w11 fk6fouc fkhj508 f1i3iumi figsok6 fpgzoln f1w7gpdv f6juhto f1gl81tg f2jf649"
          >
            Are you frustrated with finding a team? Teamtile provides these features to you😊:
          </span>
          <ul
            class="emotion-13"
          >
            <li>
              🧱View your joined teams as tiles
            </li>
            <li>
              📖View team members and channels
            </li>
            <li>
              📁Navigate to files (SharePoint Document Library)
            </li>
            <li>
              📅Navigate to Channel Calendar
            </li>
            <li>
              🪄Filter teams, channels, members
            </li>
            <li>
              📌Pin favorite teams
            </li>
          </ul>
        </section>
        <footer
          class="emotion-14"
        >
          <button
            class="fui-Link ___52mpgl0_k1g75g0 f2hkw1w f3rmtva f1ewtqcl fyind8e f1k6fduh f1w7gpdv fk6fouc fkhj508 figsok6 f1s184ao f1mk8lai fnbmjn9 f1o700av f1iuv45f f1cmlufx f9n3di6 f1ids18y f1tx3yz7 f1deo86v f1eh06m1 f1iescvh fhgqx19 f1olyrje f1p93eir f1nev41a f1ern45e f1n71otn f1h8hb77 f1deefiw"
            type="button"
          >
            Terms of Use
          </button>
          <span
            class="fui-Text emotion-15 ___c56hd20_1xy0w11 fk6fouc fkhj508 f1i3iumi figsok6 fpgzoln f1w7gpdv f6juhto f1gl81tg f2jf649"
          >
            |
          </span>
          <button
            class="fui-Link ___52mpgl0_k1g75g0 f2hkw1w f3rmtva f1ewtqcl fyind8e f1k6fduh f1w7gpdv fk6fouc fkhj508 figsok6 f1s184ao f1mk8lai fnbmjn9 f1o700av f1iuv45f f1cmlufx f9n3di6 f1ids18y f1tx3yz7 f1deo86v f1eh06m1 f1iescvh fhgqx19 f1olyrje f1p93eir f1nev41a f1ern45e f1n71otn f1h8hb77 f1deefiw"
            type="button"
          >
            Privacy Policy
          </button>
        </footer>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
